File: issue-tracker-gateway/controllers/issueController.py
from fastapi import APIRouter, Header
from models.schemas import CreateIssueSchema, AssignIssueSchema, UpdateStatusSchema
import httpx

router = APIRouter(prefix="/issueservice")
SPRING_URL = "http://localhost:8001"

@router.post("/create")
async def create_issue(I: CreateIssueSchema, Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.post(f"{SPRING_URL}/issue/create", json=I.model_dump(), headers={"Token": Token})
    return response.json()

@router.get("/myissues")
async def get_my_issues(Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.get(f"{SPRING_URL}/issue/myissues", headers={"Token": Token})
    return response.json()

@router.get("/allissues")
async def get_all_issues(Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.get(f"{SPRING_URL}/issue/allissues", headers={"Token": Token})
    return response.json()

@router.put("/assign/{ID}")
async def assign_issue(ID: int, A: AssignIssueSchema, Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.put(f"{SPRING_URL}/issue/assign/{ID}", json=A.model_dump(), headers={"Token": Token})
    return response.json()

@router.get("/assignedtome")
async def get_assigned_to_me(Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.get(f"{SPRING_URL}/issue/assignedtome", headers={"Token": Token})
    return response.json()

@router.put("/updatestatus/{ID}")
async def update_status(ID: int, S: UpdateStatusSchema, Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.put(f"{SPRING_URL}/issue/updatestatus/{ID}", json=S.model_dump(), headers={"Token": Token})
    return response.json()

@router.put("/close/{ID}")
async def close_issue(ID: int, Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.put(f"{SPRING_URL}/issue/close/{ID}", headers={"Token": Token})
    return response.json()

@router.get("/getissue/{ID}")
async def get_issue(ID: int, Token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.get(f"{SPRING_URL}/issue/getissue/{ID}", headers={"Token": Token})
    return response.json()

File: IssueTracking_And_ResolutionDashboard/src/main/java/issue_tracker/controller/IssueController.java
package issue_tracker.controller;


import issue_tracker.services.IssueService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import java.util.Map;

@RestController
@RequestMapping("/issue")
@CrossOrigin("*")
public class IssueController {

    @Autowired
    IssueService IS;

    // USER creates an issue
    @PostMapping("/create")
    public Object createIssue(@RequestBody Map<String, Object> data,
                              @RequestHeader("Token") String token) {
        return IS.createIssue(data, token);
    }

    // USER sees their own issues
    @GetMapping("/myissues")
    public Object getMyIssues(@RequestHeader("Token") String token) {
        return IS.getMyIssues(token);
    }

    // MANAGER sees all issues
    @GetMapping("/allissues")
    public Object getAllIssues(@RequestHeader("Token") String token) {
        return IS.getAllIssues(token);
    }

    // MANAGER assigns issue to developer
    @PutMapping("/assign/{ID}")
    public Object assignIssue(@PathVariable("ID") Long issueId,
                              @RequestBody Map<String, Object> data,
                              @RequestHeader("Token") String token) {
        return IS.assignIssue(issueId, data, token);
    }

    // DEVELOPER sees their assigned issues
    @GetMapping("/assignedtome")
    public Object getMyAssignedIssues(@RequestHeader("Token") String token) {
        return IS.getMyAssignedIssues(token);
    }

    // DEVELOPER updates issue status
    @PutMapping("/updatestatus/{ID}")
    public Object updateStatus(@PathVariable("ID") Long issueId,
                               @RequestBody Map<String, Object> data,
                               @RequestHeader("Token") String token) {
        return IS.updateIssueStatus(issueId, data, token);
    }

    // MANAGER closes an issue
    @PutMapping("/close/{ID}")
    public Object closeIssue(@PathVariable("ID") Long issueId,
                             @RequestHeader("Token") String token) {
        return IS.closeIssue(issueId, token);
    }

    // Anyone logged in can view a single issue
    @GetMapping("/getissue/{ID}")
    public Object getIssueById(@PathVariable("ID") Long issueId,
                               @RequestHeader("Token") String token) {
        return IS.getIssueById(issueId, token);
    }
}